src/MyTested.AspNetCore.Mvc.EntityFrameworkCore/ComponentBuilderEntityFrameworkCoreExtensions.cs

```diff
@@ -59,17 +59,92 @@ public static class ComponentBuilderEntityFrameworkCoreExtensions
         /// Instance of <see cref="IBaseTestBuilderWithComponentBuilder{TBuilder}"/> type.
         /// </param>
         /// <param name="dbContextBuilder">
-        /// Action setting the <see cref="Microsoft.EntityFrameworkCore.DbContext"/> by using <see cref="IDbContextBuilder"/>.
+        /// Action setting the <see cref="Microsoft.EntityFrameworkCore.DbContext"/> by using <see cref="IWithDbContextBuilder"/>.
         /// </param>
         /// <returns>The same component builder.</returns>
         public static TBuilder WithData<TBuilder>(
             this IBaseTestBuilderWithComponentBuilder<TBuilder> builder,
-            Action<IDbContextBuilder> dbContextBuilder)
+            Action<IWithDbContextBuilder> dbContextBuilder)
             where TBuilder : IBaseTestBuilder
         {
             var actualBuilder = (BaseTestBuilderWithComponentBuilder<TBuilder>)builder;
 
-            dbContextBuilder(new DbContextBuilder(actualBuilder.TestContext));
+            dbContextBuilder(new WithDbContextBuilder(actualBuilder.TestContext));
+
+            return actualBuilder.Builder;
+        }
+
+        /// <summary>
+        /// Remove values from the <see cref="Microsoft.EntityFrameworkCore.DbContext"/> on the tested component.
+        /// </summary>
+        /// <typeparam name="TBuilder">Class representing ASP.NET Core MVC test builder.</typeparam>
+        /// <param name="builder">
+        /// Instance of <see cref="IBaseTestBuilderWithComponentBuilder{TBuilder}"/> type.
+        /// </param>
+        /// <param name="entities">
+        /// Remove values from the registered <see cref="Microsoft.EntityFrameworkCore.DbContext"/>.
+        /// </param>
+        /// <returns>The same component builder.</returns>
+        public static TBuilder WithoutData<TBuilder>(
```

Comment:
Let's separate the `With` and `Without` methods in separate classes. The first should go to `ComponentBuilderEntityFrameworkCoreWithExtensions.cs` and the second to `ComponentBuilderEntityFrameworkCoreWithoutExtensions.cs`. The same should apply to other `Without` methods as well.